addon/debounce-task.ts

```diff
@@ -4,12 +4,12 @@ import { assert } from '@ember/debug';
 import { registerDisposable } from './utils/disposable';
 import { IMap } from './interfaces';
 
-type PendingDebounce =
-  | {
-      debouncedTask: Function;
-      cancelId: EmberRunTimer;
-    }
-  | undefined;
+interface PendingDebounce {
+  debouncedTask: Function;
+  cancelId: EmberRunTimer;
+}
+
+type PendingDebounces = Map<string, PendingDebounce>;
```

Comment:
My preference would be to not define a type alias for something that can be represented simply by using generics. If we use this, which is not necessarily wrong, it forces the reader to scroll to the top to determine what `PendingDebounces` is. I think we can remove this in favor of simply using `Map<string, PendingDebounce>`. Thoughts?

Reply:
> it forces the reader to scroll to the top to determine what `PendingDebounces` is.

In VS Code at least, you can just hover to see the type information, and also Cmd + click it to jump to the definition.

![image](https://user-images.githubusercontent.com/834636/48090012-197b6b00-e206-11e8-9f2e-933cb63a3b21.png)

I primarily did this to avoid having to type out the same definition in the `getDebouncesDisposable` signature as shown above.

I'll still remove it, if you prefer.

Reply:
I agree that this feature in VSCode is awesome. I'm also thinking of those that use vim/emacs or view the code on github or in diffs. I just find the type definition more readable when it's inline.